### ISSUE TITLE
 Expression has changed after it was checked

### ISSUE THREAD

**Issue by juristr** [formly-js/ngx-formly#781]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request
```

**Current behavior**
Receiving the following error:

```
AppComponent.html:1 ERROR Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. Previous value: 'true'. Current value: 'false'.
    at viewDebugError (errors.js:52)
    at expressionChangedAfterItHasBeenCheckedError (errors.js:28)
    at checkBindingNoChanges (util.js:127)
    at checkNoChangesNodeInline (view.js:555)
    at checkNoChangesNode (view.js:520)
    at debugCheckNoChangesNode (services.js:542)
    at debugCheckRenderNodeFn (services.js:486)
    at Object.eval [as updateRenderer] (VM286 AppComponent.ngfactory.js:58)
    at Object.debugUpdateRenderer [as updateRenderer] (services.js:463)
    at checkNoChangesView (view.js:363)
```

See the Stackblitz example for more details.

**Minimal reproduction of the problem with instructions**

https://stackblitz.com/edit/angular-ngxformly-repeat?file=app%2Fapp.component.ts

* **Angular version:** 5.0.0

**Comment by juristr** [comment]
I'm also having a "cannot set length of undefined" issue here: https://github.com/formly-js/ngx-formly/blob/master/src/core/src/components/formly.form.ts#L122

Still have to investigate into that one.

As always, happy to help

**Comment by aitboudad** [comment]
ok thanks, I'm on it!

**Comment by juristr** [comment]
Awesome.

Related to [my previous comment](https://github.com/formly-js/ngx-formly/issues/781#issuecomment-371790121) there's for sure a potential issue, which I'm experiencing on my app's setup, but cannot repro it on a Stackblitz as of now.

```javascript
private resetForm(fields: FormlyFieldConfig[], newModel: any, modelToUpdate: any) {
    fields.forEach(field => {
      if ((field.fieldGroup && field.fieldGroup.length > 0) || field.fieldArray) {
        let newFieldModel = this.fieldModel(field, newModel),
          fieldModel = this.fieldModel(field, modelToUpdate);

        if (field.fieldArray) {
          >>>>> field.fieldGroup.length = 0;
          ...
        } else {
          this.resetForm(field.fieldGroup, newFieldModel, fieldModel);
        }
      }
    });
  }
```

The problem here is that by having us add `|| field.fieldArray` in the if clause, the code potentially fails in a configuration like

```
{
   type: 'repeat',
   key: 'investments',
   fieldArray: { ... }
}
```

Because there's no `field.fieldGroup` and thus it's length cannot be set to `0`. 

![image](https://user-images.githubusercontent.com/542458/37208155-9a0a4fcc-23a0-11e8-8cf5-1a745bf5e011.png)

![image](https://user-images.githubusercontent.com/542458/37208175-a29f2720-23a0-11e8-9e90-497d975e6128.png)


What exactly is that line supposed to do?

**Comment by aitboudad** [comment]
> What exactly is that line supposed to do?

clear the fieldGroup content without losing its reference, in order to create a new one based on model content.

it should be fixed by #783

**Comment by aitboudad** [comment]
released under `2.0.0-rc.37`

**Comment by juristr** [comment]
@aitboudad you rock man. Will check at latest when back in the office monday morning. have a nice WE
